src/app/clusters/ota-requestor/ota-requestor.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/* This file contains the declarations for the Matter OTA Requestor implementation and API.
+ * Applications implementing the OTA Requestor functionality must include this file.
+ */
+
+#include <ota-requestor-driver.h>
+#include <ota-requestor-interface.h>
+
+#pragma once
+
+// This class implements all of the core logic of the OTA Requestor
+class OTARequestor : public OTARequestorInterface
+{
+public:
+    // Application interface declarations start
+
+    // Application directs the Requestor to start the Image Query process
+    // and download the new image if available
+    void TriggerImmediateQuery();
```

Comment:
should this not be part of `OTARequestorInterface`? If handling command responses is handled by `OTARequestorInterface` then I would think sending a QueryImage command would be too.

I think we also at least need a method for sending `NotifyUpdateApplied` similar to `TriggerImmediateQuery`

Reply:
The idea of OTARequestorInterface is to be the minimum of what's required to bypass our implementation is substitute your own. If you are writing your own implementation then you may as well define and implement your own API and integration with the application ( TriggerImmediateQuery etc)